src/pipelines/stage.ts

```diff
@@ -42,16 +51,90 @@ export abstract class DataStage extends Stage {
   }
 
   addAlarm(id: string, props: AlarmProps): DataStage {
-    this.cloudwatchAlarms.push(new cloudwatch.Alarm(this, id, {
-      metric: props.metric,
-      comparisonOperator: props.comparisonOperator ?? cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
-      threshold: props.threshold ?? 5,
-      evaluationPeriods: props.evaluationPeriods ?? 1,
-    }));
+    this.cloudwatchAlarms.push(
+      new cloudwatch.Alarm(this, id, {
+        metric: props.metric,
+        comparisonOperator:
+          props.comparisonOperator ??
+          cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
+        threshold: props.threshold ?? 5,
+        evaluationPeriods: props.evaluationPeriods ?? 1,
+      }),
+    );
     return this;
   }
 }
 
+export abstract class StateMachineStage extends DataStage {
+  readonly targets?: events.IRuleTarget[];
+  readonly eventPattern?: events.EventPattern;
+
+  public stateMachine?: sfn.StateMachine;
+  public stateMachineInput?: { [key: string]: any };
+
+  /*
+  DataStage with helper methods to simplify StateMachine stages creation.
+  */
+  constructor(scope: Construct, id: string, props: StateMachineStageProps) {
+    /*
+    Create a stage.
+    Parameters
+    ----------
+    scope : Construct
+    Scope within which this construct is defined
+    id : str
+    Identifier of the stage
+    name : Optional[str]
+    Name of the stage
+    description :  Optional[str]
+    Description of the stage
+    */
+    super(scope, id, props);
+
+    this.stateMachine = new sfn.StateMachine(this, 'State Machine', {
+      definition: this.createStateMachineSteps(),
+    });
+
+    if (props.additionalRolePolicyStatements) {
+      for (
+        var statement,
+          _pj_c = 0,
+          _pj_a = props.additionalRolePolicyStatements,
+          _pj_b = _pj_a.length;
+        _pj_c < _pj_b;
+        _pj_c += 1
+      ) {
+        statement = _pj_a[_pj_c];
+
+        this.stateMachine.addToRolePolicy(statement);
```

Comment:
This seems like a very complicated and hard to read way of writing a for-loop. You can simplify it as:
```typescript
if (props.additionalRolePolicyStatements) {
  props.additionalRolePolicyStatements.forEach(s => {
    this.stateMachine.addToRolePolicy(s);
  });
}
```